chainer/functions/normalization/decorrelated_batch_normalization.py

```diff
@@ -0,0 +1,303 @@
+
+        if g > 1:
+            gy = gy.reshape((b * g, C) + gy.shape[2:])
+            x = x.reshape((b * g, C) + x.shape[2:])
+        x_hat = x.transpose((1, 0) + spatial_axis).reshape(C, -1)
+        gy_hat = gy.transpose((1, 0) + spatial_axis).reshape(C, -1)
+        gy_hat_pca = projection.T.dot(gy_hat)
+        gx = gy_hat_pca.reshape(
+            (C, b * g) + gy.shape[2:]).transpose((1, 0) + spatial_axis)
+        if g > 1:
+            gx = gx.reshape((-1, c) + gy.shape[2:])
+        rhs = x_hat - mean[Ellipsis, None]
+        gprojection = (x_hat - rhs).T.dot(gy_hat)
+        gmean = -gx[:, 0]
+        self.retain_outputs(())
+        return gx, gmean, gprojection
+
+    def backward(self, inputs, grad_outputs):
+        # TODO(crcrpar): Implement this.
+        raise NotImplementedError('Double backward is not implemented for'
+                                  ' fixed decorrelated batch normalization.')
+
+
+def decorrelated_batch_normalization(x, **kwargs):
+    """decorrelated_batch_normalization(x, groups=16, eps2e-5,
+running_mean=None, running_projection=None, decay=0.9)
+
+    Decorrelated batch normalization function.
+
+    It takes the input variable ``x`` and normalizes it using
+    batch statistics to make the output zero-mean and decorrelated.
+
+    Args:
+        x (:class:`~chainer.Variable`): Input variable.
+        groups (int): Number of groups to use for group whitening.
+        eps (float): Epsilon value for numerical stability.
+        running_mean (:ref:`ndarray`): Expected value of the mean. This is a
+            running average of the mean over several mini-batches using
+            the decay parameter. If ``None``, the expected mean is initialized
+            to zero.
+        running_projection (:ref:`ndarray`):
+            Expected value of the project matrix. This is a
+            running average of the projection over several mini-batches using
+            the decay parameter. If ``None``, the expected projected is
+            initialized to the identity matrix.
+        decay (float): Decay rate of moving average. It is used during
+            training.
+
+    Returns:
+        ~chainer.Variable: The output variable which has the same shape as
+        :math:`x`.
+
+    See: `Decorrelated Batch Normalization <https://arxiv.org/abs/1804.08450>`_
+
+    .. seealso:: :class:`links.DecorrelatedBatchNormalization`
+
+    """
+    groups, eps, running_mean, running_projection, decay = \
+        argument.parse_kwargs(
+            kwargs, ('groups', 16), ('eps', 2e-5), ('running_mean', None),
+            ('running_projection', None), ('decay', 0.9))
+
+    f = DecorrelatedBatchNormalization(
+        groups, eps, running_mean, running_projection, decay)
+    return f.apply((x,))[0]
+
+
+def fixed_decorrelated_batch_normalization(x, mean, projection, groups=16):
+    """fixed_decorrelated_batch_normalization(x, mean, projection, groups=16)
```

Comment:
The line is unnecessary because the signature is the same.

Reply:
Thank you!!!